woodwork/accessor_utils.py

```diff
@@ -64,29 +64,28 @@ def _update_column_dtype(series, logical_type):
         else:
             series = series.apply(_reformat_to_latlong)
 
-    if logical_type.pandas_dtype != str(series.dtype):
```

Comment:
Guessing this is being removed because, for `init_series`, we need a new series object to be created even if the dtype doesn't need to change.

With DataFrame's, though, where we're happy to keep the same series object, we should probably be cognizant of saving time where possible and not unnecessarily calling `astype`

Reply:
Yes that is correct. If you prefer, we can leave this statement in place, and we can handle the creation of the new series object inside the set_logical_type method instead.

Reply:
I think that'd make sense

Reply:
Updated